app/validators/questionnaire_schema.py

```diff
@@ -517,3 +517,11 @@ def get_answers_from_question(question):
             *question.get("dynamic_answers", {}).get("answers", []),
             *question.get("answers", []),
         ]
+
+    def get_calculated_summary_source_answer_ids(self, block_id):
```

Comment:
the function is called `get_calculated_summary_source_answer_ids` but it returns the "unit" of the first answer, so I think it should be renamed to `get_calculated_summary_unit_type`

I think you might not need this though - see other comment where you could reuse `self.get_calculation_block_ids`